pypeit/utils.py

```diff
@@ -501,20 +502,36 @@ def rebin(a, newshape):
     indices = coordinates.astype('i')  # choose the biggest smaller integer index
     return a[tuple(indices)]
 
-# TODO This function is only used by procimg.lacosmic. Can it be replaced by above?
-def rebin_evlist(frame, newshape):
-    # This appears to be from
-    # https://scipy-cookbook.readthedocs.io/items/Rebinning.html
-    shape = frame.shape
-    lenShape = len(shape)
-    factor = (np.asarray(shape)/np.asarray(newshape)).astype(int)
-    evList = ['frame.reshape('] + \
-             ['int(newshape[%d]),factor[%d],'% (i, i) for i in range(lenShape)] + \
-             [')'] + ['.sum(%d)' % (i+1) for i in range(lenShape)] + \
-             ['/factor[%d]' % i for i in range(lenShape)]
-    return eval(''.join(evList))
 
+def rebinND(img, shape):
+    """
+    Rebin a 2D image to a smaller shape. For example, if img.shape=(100,100),
+    then shape=(10,10) would take the mean of the first 10x10 pixels into a
+    single output pixel, then the mean of the next 10x10 pixels will be output
+    into the next pixel. Note that img.shape must be an integer multiple of the
+    elements in the new shape.
 
+    Args:
+        img (`numpy.ndarray`_):
+            A 2D input image
+        shape (:obj:`tuple`):
+            The desired shape to be returned. The elements of img.shape
+            should be an integer multiple of the elements of shape.
+
+    Returns:
+        img_out (`numpy.ndarray`_): The input image rebinned to shape
```

Comment:
I think the syntax is without naming the variable, at least left of the `:`

Reply:
Quite right... I'll update this!